pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -103,6 +103,77 @@ func RerunDashAlertMigration(mg *migrator.Migrator) {
 	}
 }
 
+func AddDashboardUIDPanelIDMigrations(mg *migrator.Migrator) {
+	mg.AddMigration("add dashboard_uid column to alert_rule", migrator.NewAddColumnMigration(
+		migrator.Table{Name: "alert_rule"},
+		&migrator.Column{
+			Name:     "dashboard_uid",
+			Type:     migrator.DB_NVarchar,
+			Length:   40,
+			Nullable: true,
+		},
+	))
+	mg.AddMigration("add panel_id column to alert_rule", migrator.NewAddColumnMigration(
+		migrator.Table{Name: "alert_rule"},
+		&migrator.Column{
+			Name:     "panel_id",
+			Type:     migrator.DB_BigInt,
+			Nullable: true,
+		},
+	))
+	mg.AddMigration("update dashboard_uid and panel_id from existing annotations", &updateDashboardUIDPanelIDMigration{})
+	mg.AddMigration("create index to query by dashboard_uid and panel_id", migrator.NewAddIndexMigration(
+		migrator.Table{Name: "alert_rule"},
+		&migrator.Index{
+			Name: "IDX_alert_rule_org_id_namespace_uid_dashboard_uid_panel_id",
+			Cols: []string{"org_id", "namespace_uid", "dashboard_uid", "panel_id"},
+		},
+	))
```

Comment:
the remaining migrations for the specific table are [here](https://github.com/grafana/grafana/blob/df81676d9ea9890bb9023ba5cc9f918694042165/pkg/services/sqlstore/migrations/ualert/tables.go#L155)
according to these [guidelines](https://github.com/grafana/grafana/blob/main/contribute/architecture/backend/database.md#migrations):
> Important: If there are previous migrations for a service, use that method. By adding migrations using both methods, you risk running migrations in the wrong order.